cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftRequest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.draft;
+
+
+import io.cdap.cdap.api.artifact.ArtifactSummary;
+
+import java.util.Objects;
+
+/**
+ * Request to store a draft.
+ *
+ * @param <T> Type of config that this DraftRequest contains
+ */
+public class DraftRequest<T> {
+  private final String previousHash; // currently unused but will be used later to detect collision
+  private final String name;
+  private final String description;
+  private final int revision; // currently unused but will be used later to enable draft version tracking
+  private final ArtifactSummary artifact;
+  private T config;
+
+  public DraftRequest(T config, String previousHash, String name, String description, int revision,
+                      ArtifactSummary artifact) {
+    this.config = config;
+    this.previousHash = previousHash;
+    this.name = name;
+    this.description = description;
+    this.revision = revision;
+    this.artifact = artifact;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public String getName() {
+    return name == null ? "" : name;
+  }
+
+  public int getRevision() {
+    return revision;
+  }
+
+  public ArtifactSummary getArtifact() {
+    return artifact;
+  }
+
+
+  public T getConfig() {
+    return config;
+  }
+
+  public String getPreviousHash() {
+    return previousHash;
+  }
+
+  public void clearConfig() {
+    this.config = null;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    DraftRequest<T> that = (DraftRequest<T>) o;
+    return Objects.equals(config, that.config) &&
+      Objects.equals(previousHash, that.previousHash) &&
+      Objects.equals(name, that.name) &&
+      Objects.equals(artifact, that.artifact) &&
+      revision == that.revision;
+  }
+
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(config, previousHash, revision, artifact, name, description);
+  }
+
+  /**
+   * Builder for creating DraftRequest
+   *
+   * @param <T> Type of pipeline for the DraftRequest being created
+   */
+  public static class Builder<T> {
```

Comment:
Why need a builder? Is this class mainly for ser-de purpose?

Reply:
Yes, this is more convenient for the deserializer. It's not strictly required.